### PR TITLE
Remove JournalsHelper

### PR DESCRIPTION
**Pull request by oliverguenther**
It's used only for meetings without editing functionality. Its exposed there in https://github.com/finnlabs/openproject-meeting/pull/119

Extracted from Prototype removal https://github.com/opf/openproject/pull/4323

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://oliverguenther-housekeeping-dead-code.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-housekeeping-dead-code.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

File: app/controllers/journals_controller.rb [hunk 1]
```diff
@@ -110,17 +84,6 @@ def diff
...
-      format.any(:html, :js) do
-        render locals: { journal: @journal },
-               layout: false
-      end
-    end
-  end
-
   private
 
   def find_journal
```
Comment (ulferts) [inline app/controllers/journals_controller.rb:89]: I think there are a couple of methods in here that can also be removed:
- find_journal
- valid_vield?
- default_breadcrumb
- valid_diff?

In addition to that, some of the before filters can also be removed as can the `require 'diff'`

Comment (ulferts) [inline app/controllers/journals_controller.rb:89]: Wait a sec, I overlooked that the `diff` method is still used. So you can remove less of what I wrote but you can still remove some of it.

File: config/routes.rb [hunk 2]
```diff
@@ -163,9 +163,7 @@
 
   # only providing routes for journals when there are multiple subclasses of journals
   # all subclasses will look for the journals routes
-  resources :journals, only: [:edit, :update] do
-    get :preview, on: :member
-  end
+  get '/journals' => 'journals#index'
```
Comment (ulferts) [inline config/routes.rb]: Only for consistency sake I would prefer

`resources :journals, only: :index`

File: app/controllers/journals_controller.rb (hunk 1 above)
Comment (oliverguenther) [inline app/controllers/journals_controller.rb:89]: I've rewritten the routes (and integrates the diff route while doing so), and removed `default_breadcrumb`, as the other methods are still required.

**Pull request merged**